clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -790,9 +789,108 @@ void CIRGenModule::setNonAliasAttributes(GlobalDecl GD, mlir::Operation *GO) {
   assert(!cir::MissingFeatures::setTargetAttributes());
 }
 
+static llvm::SmallVector<uint64_t> indexesOfArrayAttr(mlir::ArrayAttr indexes) {
+  llvm::SmallVector<uint64_t> inds;
+
+  for (auto i : indexes) {
+    auto ind = dyn_cast<mlir::IntegerAttr>(i);
+    assert(ind);
+    inds.push_back(ind.getValue().getSExtValue());
+  }
+
+  return inds;
+}
+
+// handle global view on the struct type
+static GlobalViewAttr createNewGlobalView(CIRGenModule &CGM, GlobalOp newGlob,
```

Comment:
The 3 different functions with the same name might affect readability, how about `createGlobalViewFromStruct`?